kotlinx-collections-immutable/src/main/kotlin/kotlinx/collections/immutable/extensions.kt

```diff
@@ -29,140 +29,333 @@ import kotlinx.collections.immutable.implementations.persistentOrderedSet.Persis
 public fun <K, V> PersistentMap<out K, V>.putAll(pairs: Array<out Pair<K, V>>): PersistentMap<K, V>
         = mutate { it.putAll(pairs) }
-
+/**
+ * Creates a new instance of the persistent map by replacing or adding entries to this persistent map
+ * from the specified key-value pairs.
+ */
 public fun <K, V> PersistentMap<out K, V>.putAll(pairs: Sequence<Pair<K, V>>): PersistentMap<K, V>
         = mutate { it.putAll(pairs) }
 
 
+/**
+ * Returns a new instance of the persistent map with the specified [key] and its corresponding value removed.
+ * If there is no entry with the specified key, this instance is returned.
+ */
 public operator fun <K, V> PersistentMap<out K, V>.minus(key: K): PersistentMap<K, V>
         = (this as PersistentMap<K, V>).remove(key)
-
+/**
+ * Returns a new instance of the persistent map with the specified [keys] and corresponding values removed.
+ */
 public operator fun <K, V> PersistentMap<out K, V>.minus(keys: Iterable<K>): PersistentMap<K, V>
         = mutate { it.minusAssign(keys) }
-
+/**
+ * Returns a new instance of the persistent map with the specified [keys] and corresponding values removed.
+ */
 public operator fun <K, V> PersistentMap<out K, V>.minus(keys: Array<out K>): PersistentMap<K, V>
         = mutate { it.minusAssign(keys) }
-
+/**
+ * Returns a new instance of the persistent map with the specified [keys] and corresponding values removed.
+ */
 public operator fun <K, V> PersistentMap<out K, V>.minus(keys: Sequence<K>): PersistentMap<K, V>
         = mutate { it.minusAssign(keys) }
 
 
+/**
+ * Returns a new persistent list of the specified elements.
+ */
 fun <E> persistentListOf(vararg elements: E): PersistentList<E> = persistentVectorOf<E>().addAll(elements.asList())
+/**
+ * Returns an empty persistent list.
+ */
 fun <E> persistentListOf(): PersistentList<E> = persistentVectorOf()
 
+
+/**
+ * Returns a new persistent set with the given elements.
+ * Elements of the returned set are iterated in the order they were specified in [elements].
+ */
 fun <E> persistentSetOf(vararg elements: E): PersistentSet<E> = PersistentOrderedSet.emptyOf<E>().addAll(elements.asList())
+/**
+ * Returns an empty persistent set.
+ */
 fun <E> persistentSetOf(): PersistentSet<E> = PersistentOrderedSet.emptyOf<E>()
 
+
+/**
+ * Returns a new persistent set with the given elements.
+ * Order of the elements in the returned set is unspecified.
+ */
 fun <E> persistentHashSetOf(vararg elements: E): PersistentSet<E> = PersistentHashSet.emptyOf<E>().addAll(elements.asList())
 
+
+/**
+ * Returns a new persistent map with the specified key-value pairs.
+ * Elements of the returned set are iterated in the order they were specified in [pairs].
+ */
 fun <K, V> persistentMapOf(vararg pairs: Pair<K, V>): PersistentMap<K, V> = PersistentOrderedMap.emptyOf<K,V>().mutate { it += pairs }
+/**
+ * Returns a new persistent map with the specified key-value pairs.
+ * Order of the elements in the returned set is unspecified.
```

Comment:
"Entries of the map are iterated in the order they were specified."

Take a look at `mapOf`, `linkedMapOf` docs, how this clause is placed there.

Reply:
+1